python/shotgun_data/shotgun_data_retriever.py

```diff
@@ -596,29 +680,51 @@ def _download_url(self, file_path, url, entity_type, entity_id, field):
         :param str field: The name of the field that contains the url. If
                           the url needs to be requeried, this field will be
                           where the fresh url is pulled from.
+        :returns: Full path the downloaded file. This value may be different
+                  than the input `file_path` if the resolved url's extension
+                  differed from what was specified.
         """
         try:
+            download_path = sgtk.util.download_url(
+                self._bundle.shotgun, url, file_path, True
+            )
+            file_path = download_path
+        except TypeError:
+            # Probably an older version of core that doesn't have the
```

Comment:
i am not 100% sure, but looks like the exception bubbling is not right - can you double check?

before, it was

```
try:
    sgtk.util.download_url(self._bundle.shotgun, url, file_path)
except TankError, e:
    do stuff
```

now it is 

```
try:
    download_path = sgtk.util.download_url
except TypeError:
    sgtk.util.download_url(self._bundle.shotgun, url, file_path)
except TankError, e:
    do stuff
```

Shouldn't it be nested, like this? 

```
try:

   try:
        download_path = sgtk.util.download_url
    except TypeError:
        sgtk.util.download_url(self._bundle.shotgun, url, file_path)

except TankError, e:
    do stuff
```

In the current implementation, what happens if `sgtk.util.download_url(self._bundle.shotgun, url, file_path)` raises a TankError?